From d2cd5e641efe08a4f5f467ec16297a704a391e81 Mon Sep 17 00:00:00 2001
From: Stefan Liebler <stli@linux.vnet.ibm.com>
Date: Mon, 7 Nov 2016 16:06:01 +0100
Subject: [PATCH 07/17] S390: Fix remaining ONE_DIRECTION warning messages

upstream commit f349489e7e5b5341a9c1a590e9a41c2e07d3bdbb

This patch fixes the remaining ONE_DIRECTION warnings for s390 specific conversions.
It defines ONE_DIRECTION to 0 like the patch from Steve Ellcey:
https://www.sourceware.org/ml/libc-alpha/2014-05/msg00039.html

    Changelog:
    * sysdeps/s390/s390-64/utf16-utf32-z9.c
    (ONE_DIRECTION): Define.
    * sysdeps/s390/s390-64/utf8-utf16-z9.c
    (ONE_DIRECTION): Define.
    * sysdeps/s390/s390-64/utf8-utf32-z9.c
    (ONE_DIRECTION): Define.
---
 sysdeps/s390/s390-64/utf16-utf32-z9.c | 1 +
 sysdeps/s390/s390-64/utf8-utf16-z9.c  | 1 +
 sysdeps/s390/s390-64/utf8-utf32-z9.c  | 1 +
 3 files changed, 3 insertions(+)

diff --git a/sysdeps/s390/s390-64/utf16-utf32-z9.c b/sysdeps/s390/s390-64/utf16-utf32-z9.c
index 94a1a33..ddc42fe 100644
--- a/sysdeps/s390/s390-64/utf16-utf32-z9.c
+++ b/sysdeps/s390/s390-64/utf16-utf32-z9.c
@@ -44,6 +44,7 @@
 #define FROM_LOOP		from_utf16_loop
 #define TO_LOOP			to_utf16_loop
 #define FROM_DIRECTION		(dir == from_utf16)
+#define ONE_DIRECTION           0
 #define PREPARE_LOOP							\
   enum direction dir = ((struct utf16_data *) step->__data)->dir;	\
   int emit_bom = ((struct utf16_data *) step->__data)->emit_bom;	\
diff --git a/sysdeps/s390/s390-64/utf8-utf16-z9.c b/sysdeps/s390/s390-64/utf8-utf16-z9.c
index 8e0515c..2c2d92c 100644
--- a/sysdeps/s390/s390-64/utf8-utf16-z9.c
+++ b/sysdeps/s390/s390-64/utf8-utf16-z9.c
@@ -42,6 +42,7 @@
 #define FROM_LOOP		from_utf8_loop
 #define TO_LOOP			to_utf8_loop
 #define FROM_DIRECTION		(dir == from_utf8)
+#define ONE_DIRECTION           0
 #define PREPARE_LOOP							\
   enum direction dir = ((struct utf8_data *) step->__data)->dir;	\
   int emit_bom = ((struct utf8_data *) step->__data)->emit_bom;		\
diff --git a/sysdeps/s390/s390-64/utf8-utf32-z9.c b/sysdeps/s390/s390-64/utf8-utf32-z9.c
index c657a38..c582155 100644
--- a/sysdeps/s390/s390-64/utf8-utf32-z9.c
+++ b/sysdeps/s390/s390-64/utf8-utf32-z9.c
@@ -44,6 +44,7 @@
 #define FROM_LOOP		from_utf8_loop
 #define TO_LOOP			to_utf8_loop
 #define FROM_DIRECTION		(dir == from_utf8)
+#define ONE_DIRECTION           0
 #define PREPARE_LOOP							\
   enum direction dir = ((struct utf8_data *) step->__data)->dir;	\
   int emit_bom = ((struct utf8_data *) step->__data)->emit_bom;		\
-- 
1.8.3.1

